raft/src/main/java/org/apache/kafka/raft/ReplicatedCounter.java

```diff
@@ -87,6 +96,25 @@ public synchronized void handleCommit(BatchReader<Integer> reader) {
         }
     }
 
+    @Override
+    public synchronized void handleSnapshot(SnapshotReader<Integer> reader) {
+        try {
+            try (SnapshotReader<Integer> snapshot = reader) {
+                log.debug("Loading snapshot {}", snapshot.snapshotId());
+                for (List<Integer> batch : snapshot) {
```

Comment:
I guess snapshots for this replicated counter would end up with a single value?

Reply:
Yes. I missed this. `ReplicatedCounter` now generates a snapshot after more than 10 records. Fix the existing simulation tests to take snapshot into account.